packages/fxa-content-server/app/scripts/models/subscription.ts

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent current subscription state, so that metrics can
+ * associate events with products and plans.
+ *
+ * Tries to read data from the URL or, failing that, the resume token.
+ */
+
+import Backbone from 'backbone';
+import Cocktail from '../lib/cocktail';
+import ResumeTokenMixin from './mixins/resume-token';
+import UrlMixin from './mixins/url';
+
+const SUBSCRIBE_PRODUCT_PATHNAME_REGEXP = /^\/subscriptions\/products\/(prod_\w+)/;
+
+// Neither UrlMixin nor ResumeTokenMixin are classes or interfaces
+interface IUrlMixin {
```

Comment:
@bbangert what's a good way to define interfaces for JS files like these that don't export classes but are still mixins? My typescript foo is not great here.

Reply:
UrlMixin and RefreshTokenMixin are mixins, but not Typescript type mixins based on classes. Typescript needs to be made aware their interface is available on the consuming class or else it complains about methods not being available.

Reply:
I think you can just make an interface that extends the other multiple classes, then assign it.

e.g. https://www.typescriptlang.org/docs/handbook/mixins.html

Reply:
I tried that but the reference mixins are class based, ours are object based and then typescript blows up.

Reply:
Doh. I think doing this is as good as it gets then.